timemachine/cpp/src/wrap_kernels.cpp

```diff
@@ -308,6 +309,107 @@ void declare_context(py::module &m) {
+               const int burn_in,
+               const int store_x_interval,
+               const double radius,
+               const double k) -> py::tuple {
+                if (n_steps <= 0) {
+                    throw std::runtime_error("local steps must be at least one");
+                }
+                if (burn_in < 0) {
+                    throw std::runtime_error("burn in steps must be greater or equal to zero");
+                }
+                // Lower bound on radius selected to be 1 Angstrom, to avoid case where no particles
+                // are moved. TBD whether or not this is a good lower bound
+                const double min_radius = 0.1;
+                if (radius < min_radius) {
+                    throw std::runtime_error("radius must be greater or equal to " + std::to_string(min_radius));
+                }
+                if (k < 1.0) {
+                    throw std::runtime_error("k must be at least one");
+                }
+                // TBD determine a more precise threshold, currently 10x what has been tested
+                const double max_k = 1000000.0;
+                if (k > max_k) {
+                    throw std::runtime_error("k must be less than than " + std::to_string(max_k));
+                }
+
+                const int N = ctxt.num_atoms();
+                const int x_interval = (store_x_interval <= 0) ? n_steps : store_x_interval;
+
+                std::vector<int> vec_selection_mask(selection_mask.size());
+                std::memcpy(vec_selection_mask.data(), selection_mask.data(), vec_selection_mask.size() * sizeof(int));
+                verify_atom_idxs(N, vec_selection_mask);
+
+                std::array<std::vector<double>, 2> result =
+                    ctxt.multiple_steps_local_selection(n_steps, reference_idx, vec_selection_mask, burn_in, x_interval, radius, k);
+                const int D = 3;
+                const int F = result[0].size() / (N * D);
+                py::array_t<double, py::array::c_style> out_x_buffer({F, N, D});
+                std::memcpy(out_x_buffer.mutable_data(), result[0].data(), result[0].size() * sizeof(double));
+
+                py::array_t<double, py::array::c_style> box_buffer({F, D, D});
+                std::memcpy(box_buffer.mutable_data(), result[1].data(), result[1].size() * sizeof(double));
+                return py::make_tuple(out_x_buffer, box_buffer);
+            },
+            py::arg("n_steps"),
+            py::arg("reference_idx"),
+            py::arg("selection_mask"),
+            py::arg("burn_in") = 500, // This is arbitrarily selected as a default, TODO make informed choice
+            py::arg("store_x_interval") = 0,
+            py::arg("radius") = 1.2,
+            py::arg("k") = 10000.0,
+            R"pbdoc(
+        Take multiple steps using a selection of free particles restrainted to a reference particle. Useful for avoiding the bias
```

Comment:
restrainted -> restrained (also L402)

Reply:
Good catch, fixed in https://github.com/proteneer/timemachine/pull/1005/commits/c4b945361803938a86533a84cc6b441e666a025a